catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -535,15 +552,22 @@ private ContentItem generateContentFile(ContentItem item, Path contentDirectory)
 
         long copy;
 
-        try (InputStream inputStream = item.getInputStream()) {
-            copy = Files.copy(inputStream, contentItemPath);
-        }
+        if (storeReference != null) {
+            copy = item.getSize();
+            Files.write(Paths.get(contentItemPath.toString() + "." + REF_EXT),
+                    storeReference.getBytes(Charset.forName("UTF-8")));
+            contentItemPath = Paths.get(storeReference);
+        } else {
+            try (InputStream inputStream = item.getInputStream()) {
+                copy = Files.copy(inputStream, contentItemPath);
+            }
 
-        if (copy != item.getSize()) {
-            LOGGER.info("Created content item {} size {} does not match expected size {}",
-                    item.getId(),
-                    copy,
-                    item.getSize());
+            if (copy != item.getSize()) {
+                LOGGER.warn("Created content item {} size {} does not match expected size {}",
```

Comment:
If it is happening on every request, something is very wrong and needs to be fixed immediately. If it only happens occasionally, it will not flood the log. I think the risk is acceptable. I will add the suggestion to check filesystem/network integrity.